tests/SolrIndexTest.php

```diff
@@ -376,6 +398,94 @@ public function testGetIndexNameWithPrefixAndSuffixFromEnvironment()
         );
     }
 
+    /**
+     * Test that ShowInSearch and getShowInSearch() exclude DataObjects from being added to the index
+     *
+     * Note: this code path that really being tested here is SearchUpdateProcessor->prepareIndexes()
+     * This code path is used for 'inlet' filtering on CMS->save()
+     * The results of this will show-up in SolrIndex->_addAs()
+     */
+    public function testShowInSearch()
+    {
+        $defaultMode = Versioned::get_reading_mode();
+        Versioned::set_reading_mode('Stage.' . Versioned::DRAFT);
+
+        $serviceMock = $this->getMockBuilder(Solr4Service::class)
+            ->setMethods(['addDocument', 'deleteById'])
+            ->getMock();
+
+        $index = new SolrIndexTest_ShowInSearchIndex();
+        $index->setService($serviceMock);
+        FullTextSearch::force_index_list($index);
+
+        // will get added
+        $pageOne = new Page();
+        $pageOne->Title = 'Test Page true';
+        $pageOne->ShowInSearch = true;
+        $pageOne->write();
+
+        // will get filtered out
+        $page = new Page();
+        $page->Title = 'Test Page false';
+        $page->ShowInSearch = false;
+        $page->write();
+
+        // will get added
+        $file = new File();
+        $file->Title = 'Test File true';
+        $file->ShowInSearch = true;
+        $file->write();
+
+        // will get filtered out
+        $file = new File();
+        $file->Title = 'Test File false';
+        $file->ShowInSearch = false;
+        $file->write();
+
+        // will get added
+        $objone = new SolrIndexTest_MyDataObjectOne();
+        $objone->Title = 'Test MyDataObjectOne true';
+        $objone->ShowInSearch = true;
+        $objone->write();
+
+        // will get filtered out
+        $objone = new SolrIndexTest_MyDataObjectOne();
+        $objone->Title = 'Test MyDataObjectOne false';
+        $objone->ShowInSearch = false;
+        $objone->write();
+
+        // will get added
+        // this class has a getShowInSearch() == true, which will override $mypage->ShowInSearch = false
+        $objtwo = new SolrIndexTest_MyDataObjectTwo();
+        $objtwo->Title = 'Test MyDataObjectTwo false';
+        $objtwo->ShowInSearch = false;
+        $objtwo->write();
+
+        // will get added
+        // this class has a getShowInSearch() == true, which will override $mypage->ShowInSearch = false
+        $mypage = new SolrIndexTest_MyPage();
+        $mypage->Title = 'Test MyPage false';
+        $mypage->ShowInSearch = false;
+        $mypage->write();
+
+        // 1x Page + 1x File + 1x SolrIndexTest_MyPage,
+        // 1x SolrIndexTest_MyDataObjectOne + 1x SolrIndexTest_MyDataObjectTwo
+        $serviceMock->expects($this->exactly(5))->method('addDocument');
```

Comment:
That's a bit of a weak test, it would also pass if the first object is added five times, and the other ones ignored. I would break down test methods to a more granular level, as a general comment. These long test setups are really hard to read.

Reply:
Have implemented testing the parameters passed to the $mock->addDocument() so we no longer risk passing it when it's actually failing